codespell_lib/tests/test_basic.py

```diff
@@ -455,6 +455,30 @@ def test_context(tmpdir, capsys):
     assert 'ERROR' in lines[0]
 
 
+def test_ignore_regex_flag(tmpdir, capsys):
+    """Test ignore regex flag functionality."""
+    d = str(tmpdir)
+
+    # Invalid regex.
+    code, stdout, _ = cs.main('--ignore-regex=(', std=True)
+    assert code == EX_USAGE
+    assert 'usage:' in stdout
+
+    with open(op.join(d, 'flag.txt'), 'w') as f:
+        f.write('# Please see http://example.com/abandonned for info\n')
+    # Test file has 1 invalid entry, and it's not ignored by default.
+    assert cs.main(f.name) == 1
+    # An empty regex is the default value, and nothing is ignored.
+    assert cs.main(f.name, '--ignore-regex=') == 1
+    assert cs.main(f.name, '--ignore-regex=""') == 1
+    # Non-matching regex results in nothing being ignored.
+    assert cs.main(f.name, '--ignore-regex=^$') == 1
+    # A word can be ignored.
+    assert cs.main(f.name, '--ignore-regex=abandonned') == 0
```

Comment:
It would be good to test the real world functionality, so e.g. add a typo (I guess ideally a different one) elsewhere in the example text, then do a check of a regex match of it (e.g. with a character class) and the word boundary markers and confirm the flag actually works in reality.

Reply:
Done